src/files-breadcrumbs.tsx

```diff
@@ -122,7 +125,7 @@ function BookmarkButton({ path }: { path: string[] }) {
                 const newBoomark = path.map(part => encodeURIComponent(part))
                         .join('/');
                 if (bookmarks.includes(currentPath)) {
-                    return old_content.split('\n').filter(line => !line.startsWith(`file://${newBoomark}`))
```

Comment:
This looks like it'll work now.

A sketch for a testcase: take a "special" character that we know won't be escaped, something like `-` (which can also be written `%2d`).

Then

```sh
mkdir /home/admin/test-dir
echo file:///home/admin/test%2ddir >> [bookmarks]
```

Then verify that:
 - we can use the bookmark to navigate to the new directory (even though it has a weird encoding)
 - we can remove the bookmark
 - we can check the file to see that the bookmark has been removed, and we can check the UI to see that it's been updated as expected
 - we can add it back again
 - the encoding that it got added back with is different than the original one (ie: check that `test-dir` appears in the bookmarks file now)

Reply:
Right because:
```
decodeURIComponent("test%2ddir")
"test-dir" 
```

```
encodeURIComponent("test-dir")
"test-dir" 
```

Reply:
I just reworked the test with spaces to use %2d so this is now fully tested.